marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/builtins/steps/mapping/entity-services/lib.sjs

```diff
@@ -476,21 +476,31 @@ function getMarkLogicMappingFunctions() {
   return fn.head(datahub.hubUtils.queryLatest(function() {
     let fnMetadata = fn.collection("http://marklogic.com/entity-services/function-metadata")
     let ns = {"m":"http://marklogic.com/entity-services/mapping"};
+    const functionsAdded = [];
     let output = [];
 
     for (const metaData of fnMetadata){
       if(metaData.xpath("/m:function-defs",ns)) {
         let j = 1;
         let fnLocation = metaData.xpath("/m:function-defs/@location",ns)
         for (const mlFunction of metaData.xpath("/m:function-defs/m:function-def",ns )){
-          let funcName = metaData.xpath("/m:function-defs/m:function-def["+j+"]/@name", ns);
+          let funcName = String(metaData.xpath("/m:function-defs/m:function-def["+j+"]/@name", ns));
           let params = String(metaData.xpath("/m:function-defs/m:function-def["+j+"]/m:parameters/m:parameter/@name",ns)).replace("\n",",");
           j++;
-          let singleFunction ={};
-          singleFunction["functionName"] = funcName;
-          singleFunction["signature"] = funcName +"("+params+")";
-          singleFunction["category"] = (String(fnLocation).includes("/data-hub/5/mapping-functions")) ? "builtin" : "custom";
-          output.push(singleFunction)
+          if(functionsAdded.includes(funcName)){
+            let addedFunction = output.find(func => {
+              return func.functionName == funcName
+            });
```

Comment:
Instead of searching in both `functionsAdded` and `output` arrays, can we use a Map here with the key being the function name and the value being the function object?
Map also provides an iterator for values that can then be used to create the output array.
```
const map1 = new Map();
map1.set('a', {'a':'a'});
map1.set('b', {'b':'b'});
const ar = [...map1.values()];
```

Reply:
That's a good call, that will be a lot faster than calling "includes" N times and then having to call "find" as well.

Reply:
The spread operator doesn't work in ML 9. So, iterated through the values.